source/includes/api/_metadata.md

```diff
@@ -1,188 +1,436 @@
+
+Credentials required for creating, updating or deleting:
+
+- You must have an `ADMIN` or `MANAGER` role.
+- You must belong to the same `application` group as the dataset associated with the metadata object.
+
+> Overview of endpoints
+
+| Method | Path                                | Description                                                                                                                       |
+|--------|-------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|
+| POST   | `v1/dataset/<dataset-id>/metadata`                      | [Create a dataset metadata object](#create-a-metadata-object) |
+| POST   | `v1/dataset/<dataset-id>/layer/<layer-id>/metadata` | [Create a dataset layer metadata object](#create-a-metadata-object) |
+| POST   | `v1/dataset/<dataset-id>/widget/<widget-id>/metadata` | [Create a dataset widget metadata object](#create-a-metadata-object) |
+| GET   | `v1/metadata` | [Get all metadata objects](#getting-all-metadata-objects) |
+| POST   | `v1/dataset/:dataset/metadata/find-by-ids` | [Get metadata objects by identifiers](#getting-metadata-objects-by-identifiers) |
+| POST   | `v1/dataset/:dataset/layer/<layer_id>/metadata/find-by-ids` | [Get metadata objects by identifiers](#getting-metadata-objects-by-identifiers) |
+| POST   | `v1/dataset/:dataset/widget/<widget_id>/metadata/find-by-ids` | [Get metadata objects by identifiers](#getting-metadata-objects) |
+| GET   | `v1/dataset/:dataset/metadata` | [Get metadata objects](#getting-metadata-objects) |
+| GET   | `v1/dataset/:dataset/layer/<layer_id>/metadata` | [Get metadata objects](#getting-metadata-objects-by-identifiers) |
+| GET   | `v1/dataset/:dataset/widget/<widget_id>/metadata` | [Get metadata objects](#getting-metadata-objects) |
+| UPDATE   | `v1/dataset/:dataset/metadata` | [Get metadata objects](#getting-metadata-objects) |
+| UPDATE   | `v1/dataset/:dataset/layer/<layer_id>/metadata` | [Get metadata objects](#getting-metadata-objects-by-identifiers) |
+| UPDATE   | `v1/dataset/:dataset/widget/<widget_id>/metadata` | [Get metadata objects](#getting-metadata-objects) |
+
+## Create a metadata object
+
+To associate a metadata object with a Dataset, and it's Layers or Widgets you need to provide your authorization token (`<auth_token>`) in the header, and POST a JSON object that must contain valid values for the fields `application` and `language` to the appropriate `/dataset/<dataset_id>` endpoint. If the object is correctly added to the entity a `200` response is returned, as well as the new metadata object. If the user is not authorized to create metadata objects a `403` error is returned; see the required [credentials](#overview-of-available-endpoints).
```

Comment:
See my comment on the credentials for creating, updating, and deleting metadata - having a token is not enough to perform those actions. The user also has to have the role `ADMIN` or `MANAGER`, which means that `USER` role (the default user role in the API!) users cannot create metadata. _(maybe this is something we want to highlight somewhere in the intro sections, I find this very important for "common" API users that want to create metadata)_

I'd add a section below exclusively to detail the possible errors that can occur while creating metadata. And I'd refer these details there, along with the error codes/messages when we don't provide the required fields (which I believe we are missing in the docs)